redis-util/src/main/java/org/sunbird/redis/RedisConnectionManager.java

```diff
@@ -0,0 +1,143 @@
+  private static String port = propertiesCache.getProperty("sunbird_redis_port");
+  private static RedissonClient client = null;
+  private static final String KEY_SPACE_NAME = "sunbird";
+  private static final String PAGE_MANAGEMENT_TABLE_NAME = "page_management";
+  private static final String PAGE_SECTION_TABLE_NAME = "page_section";
+  private static RedisUtil redisUtil = new RedisUtil();
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+
+  static {
+    initialiseConnection();
+    initializePageCache();
+    initializeSectionCache();
+  }
+
+  public static RedissonClient getClient() {
+    if (client == null) {
+      ProjectLogger.log("Redis  clinet is null ", LoggerEnum.INFO.name());
+      initialiseConnection();
+      ProjectLogger.log("After calling initialiseConnection Redis client value " + client, LoggerEnum.INFO.name());
+    }
+    return client;
+  }
+
+  private static boolean initialiseConnection() {
+    try {
+      if (isRedisCluster) {
+        ProjectLogger.log("RedisConnectionManager:initialiseConnection initalising Cluster.", LoggerEnum.INFO.name());
+        initialisingClusterServer(host, port);
+        return true;
+      }
+      ProjectLogger.log("RedisConnectionManager:initialiseConnection initalising Single Server.",
+          LoggerEnum.INFO.name());
+      initialiseSingleServer(host, port);
+      return true;
+    } catch (Exception e) {
+      ProjectLogger.log("Error while initialising elastic search connection", e);
+      return false;
+    }
+  }
+
+  private static void initialiseSingleServer(String host, String port) {
+    Config config = new Config();
+    config.useSingleServer().setAddress(host + ":" + port);
+    config.setCodec(new StringCodec());
+    client = Redisson.create(config);
+
+  }
+
+  // cluster state scan interval in milliseconds
+  // use "rediss://" for SSL connection
+  private static void initialisingClusterServer(String host, String port) {
+    String[] hosts = host.split(",");
+    String[] ports = port.split(",");
+    Config config = new Config();
+    try {
+      config.setCodec(new StringCodec());
+      ClusterServersConfig clusterConfig = config.useClusterServers();
+      ProjectLogger.log("RedisConnectionManager:initialisingClusterServer with ip :" + hosts.toString(),
+          LoggerEnum.INFO.name());
+
+      clusterConfig.setScanInterval(2000);
+      for (int i = 0; i < hosts.length && i < ports.length; i++) {
+        clusterConfig.addNodeAddress("redis://" + hosts[i] + ":" + ports[i]);
+      }
+
+      client = Redisson.create(config);
+      ProjectLogger.log("RedisConnectionManager:initialisingClusterServer client created ", LoggerEnum.INFO.name());
+    } catch (Exception e) {
+      ProjectLogger.log("RedisConnectionManager:initialisingClusterServer Exception " + e, LoggerEnum.INFO.name());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public static void initializeSectionCache() {
```

Comment:
RedisCache implementation should be agnostic of the usage. SectionCache specific logic should be part of page management service / actor.